services/auth/signin.go

```diff
@@ -23,31 +21,26 @@ import (
 
 // UserSignIn validates user name and password.
 func UserSignIn(username, password string) (*models.User, *login.Source, error) {
+	trimmedUsername := strings.TrimSpace(username)
+	if len(trimmedUsername) == 0 {
+		return nil, nil, models.ErrUserNotExist{Name: trimmedUsername}
+	}
+
 	var user *models.User
-	if strings.Contains(username, "@") {
-		user = &models.User{Email: strings.ToLower(strings.TrimSpace(username))}
-		// check same email
-		cnt, err := db.Count(user)
+	var hasUser bool
+	var err error
+	if strings.Contains(trimmedUsername, "@") { // login with activated email address
+		user, err = models.GetUserByEmail(trimmedUsername)
 		if err != nil {
 			return nil, nil, err
 		}
-		if cnt > 1 {
-			return nil, nil, user_model.ErrEmailAlreadyUsed{
-				Email: user.Email,
-			}
-		}
+		hasUser = true
 	} else {
-		trimmedUsername := strings.TrimSpace(username)
-		if len(trimmedUsername) == 0 {
-			return nil, nil, models.ErrUserNotExist{Name: username}
-		}
-
 		user = &models.User{LowerName: strings.ToLower(trimmedUsername)}
-	}
-
-	hasUser, err := models.GetUser(user)
-	if err != nil {
-		return nil, nil, err
+		hasUser, err = models.GetUser(user)
```

Comment:
Do we need a `GetUserByName` like `GetUserByEmail`?

Then the username can be prepared (trimed, lower-cased) correctly inside this function.

Reply:
I think we already have `GetUserByName`.

Reply:
Why don't we use it?

Reply:
done.